recipe_scrapers/thecookingguy.py

```diff
@@ -31,9 +28,10 @@ def title(self):
         return self.schema.title()
 
     def total_time(self):
-        warnings.warn(
-            null_return_warning.format(self.host(), "total_time", BUG_REPORT_LINK)
-        )
+        if sys.warnoptions:
+            warnings.warn(
+                null_return_warning.format(self.host(), "total_time", BUG_REPORT_LINK)
+            )
```

Comment:
This is slightly different to my suggestion, I think; I reckon we should always emit the warning (no `if` condition gating it), but because I'm wary of people copy-and-pasting code, I don't think we should include a `warnings.simplefilter` line that ignores _all_ warnings (even -- or perhaps especially -- when it is only enabled if `sys.warnoptions` is set).

Reply:
oh you're right, I'll go update it

Reply:
No problem :)  thanks!

Reply:
Or wait no, I think we were using the ignore filter to ignore warnings by default when no flags are set. I can change it to just follow the default behavior of the flag and always emitting with no flag if you want?

Reply:
I think I probably overcomplicated the requirements, to be honest :/

Let's begin by emitting a warning -- without applying any conditional filtering -- and see what the results of that are like.

Reply:
Without any conditional filtering, the warning always gets emitted except when the -Wignore flag is used.

Reply:
Alright, I think that's acceptable; perhaps an improvement - and it's simpler to reason about.  Does that seem OK to you too?

Reply:
Something that could be done in the future is creating a centralized custom warnings set:
https://angwalt12.medium.com/custom-warnings-in-python-aa61022c725c